lib/icinga/checkable-check.cpp

```diff
@@ -477,7 +477,7 @@ void Checkable::ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrig
 
 	if (send_notification && !is_flapping) {
 		if (!IsPaused()) {
-			if (suppress_notification) {
+			if (suppress_notification || (GetSuppressedNotifications() & (NotificationRecovery|NotificationProblem))) {
```

Comment:
No, not neutralize but rather keep the suppression ongoing until suppressed notifications are fired. This is needed because otherwise, a notification might be sent because there was a state change compared to the last state within the suppression period.

But I should probably add a comment here.

Reply:
Do this...